examples/counter/containers/App.js

```diff
@@ -1,16 +1,11 @@
 import React, { Component } from 'react';
-import CounterApp from './CounterApp';
-import { createStore, applyMiddleware, combineReducers } from 'redux';
-import thunk from 'redux-thunk';
 import { Provider } from 'react-redux';
-import * as reducers from '../reducers';
-
-const createStoreWithMiddleware = applyMiddleware(thunk)(createStore);
-const reducer = combineReducers(reducers);
-const store = createStoreWithMiddleware(reducer);
+import CounterApp from './CounterApp';
+import createCounterStore from '../store/createCounterStore';
 
 export default class App extends Component {
   render() {
+    const store = createCounterStore(this.props.initialState);
     return (
       <Provider store={store}>
         {() => <CounterApp />}
```

Comment:
Maybe it's easier to test just the `CounterApp` from the unit tests? Then you don't need `this.props.initialState`, and you can just render `<Provider store={testStore}>{() => <CounterApp />}></Provider>` in your tests.